torch/testing/_creation.py

```diff
@@ -0,0 +1,159 @@
+    _floating_types = [torch.float16, torch.bfloat16, torch.float32, torch.float64]
+    _complex_types = [torch.cfloat, torch.cdouble]
+
+    if dtype is torch.bool:
+        result = torch.randint(0, 2, size, device=device, dtype=dtype)
+    elif dtype is torch.uint8:
+        ranges = (torch.iinfo(dtype).min, torch.iinfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges[0], ranges[1], 0, 9, dtype)
+        result = torch.randint(low, high, size, device=device, dtype=dtype)
+    elif dtype in _integral_types:
+        ranges = (torch.iinfo(dtype).min, torch.iinfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges[0], ranges[1], -9, 9, dtype)
+        result = torch.randint(low, high, size, device=device, dtype=dtype)
+    elif dtype in _floating_types:
+        ranges_floats = (torch.finfo(dtype).min, torch.finfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges_floats[0], ranges_floats[1], -9, 9, dtype)
+        rand_val = torch.rand(size, device=device, dtype=dtype)
+        result = high * rand_val + low * (1 - rand_val)
+    elif dtype in _complex_types:
+        float_dtype = torch.float if dtype is torch.cfloat else torch.double
+        ranges_floats = (torch.finfo(float_dtype).min, torch.finfo(float_dtype).max)
+        low, high = _modify_low_high(low, high, ranges_floats[0], ranges_floats[1], -9, 9, dtype)
+        real_rand_val = torch.rand(size, device=device, dtype=float_dtype)
+        imag_rand_val = torch.rand(size, device=device, dtype=float_dtype)
+        real = high * real_rand_val + low * (1 - real_rand_val)
+        imag = high * imag_rand_val + low * (1 - imag_rand_val)
+        result = torch.complex(real, imag)
+    else:
+        raise TypeError(f"Given dtype {dtype} is not supported for make_tensor, if you think"
+                        " that it should be supported, please file an issue"
+                        " here: https://github.com/pytorch/pytorch/issues")
+
+    if noncontiguous and result.numel() > 1:
+        result = torch.repeat_interleave(result, 2, dim=-1)
+        result = result[..., ::2]
+
+    if exclude_zero:
+        if dtype in _integral_types or dtype is torch.bool:
+            replace_with = torch.tensor(1, device=device, dtype=dtype)
+        elif dtype in _floating_types:
+            replace_with = torch.tensor(torch.finfo(dtype).tiny, device=device, dtype=dtype)
+        elif dtype in _complex_types:
+            float_dtype = torch.float if dtype is torch.cfloat else torch.double
+            float_eps = torch.tensor(torch.finfo(float_dtype).tiny, device=device, dtype=float_dtype)
+            replace_with = torch.complex(float_eps, float_eps)
+        else:
+            raise TypeError(f"Given dtype {dtype} is not supported for make_tensor, if you think"
+                            " that it should be supported, please file an issue"
+                            " here: https://github.com/pytorch/pytorch/issues")
```

Comment:
I don't think we need to ask for user feedback here. If we add a new dtype, I'm certain someone from us will use `make_tensor` and discover this. cc @mruberry

Reply:
Telling people to file an issue seems fine. I'd tweak the wording to be:

"The requested dtype DTYPE is not supported BY torch.testing.make_tensor(). To request support file an issue at: LINK."